pkg/operator/controllers/pullsecret/pullsecret_controller.go

```diff
@@ -142,50 +139,27 @@ func (r *PullSecretReconciler) pullsecret(ctx context.Context) (*corev1.Secret,
 	return ps, false, nil
 }
 
-func triggerReconcile(secret *corev1.Secret) bool {
-	return (secret.Name == pullSecretName.Name && secret.Namespace == pullSecretName.Namespace) ||
-		(secret.Name == operator.SecretName && secret.Namespace == operator.Namespace)
-}
-
 // SetupWithManager setup our manager
 func (r *PullSecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	isPullSecret := predicate.Funcs{
-		UpdateFunc: func(e event.UpdateEvent) bool {
-			_, ok := e.ObjectOld.(*arov1alpha1.Cluster)
-			if ok {
-				return true
-			}
-
-			secret, ok := e.ObjectOld.(*corev1.Secret)
-			return ok && triggerReconcile(secret)
-		},
-		CreateFunc: func(e event.CreateEvent) bool {
-			_, ok := e.Object.(*arov1alpha1.Cluster)
-			if ok {
-				return true
-			}
-
-			secret, ok := e.Object.(*corev1.Secret)
-			return ok && triggerReconcile(secret)
-		},
-		DeleteFunc: func(e event.DeleteEvent) bool {
-			_, ok := e.Object.(*arov1alpha1.Cluster)
-			if ok {
-				return true
-			}
+	pullSecretPredicate := predicate.NewPredicateFuncs(func(meta metav1.Object, object runtime.Object) bool {
+		return (meta.GetName() == pullSecretName.Name && meta.GetNamespace() == pullSecretName.Namespace) ||
+			(meta.GetName() == operator.SecretName && meta.GetNamespace() == operator.Namespace)
+	})
 
-			secret, ok := e.Object.(*corev1.Secret)
-			return ok && triggerReconcile(secret)
-		},
-	}
+	aroClusterPredicate := predicate.NewPredicateFuncs(func(meta metav1.Object, object runtime.Object) bool {
+		return meta.GetName() == arov1alpha1.SingletonClusterName
+	})
 
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&arov1alpha1.Cluster{}).
+		For(&arov1alpha1.Cluster{}, builder.WithPredicates(aroClusterPredicate)).
 		// https://github.com/kubernetes-sigs/controller-runtime/issues/1173
 		// equivalent to For(&v1.Secret{})., but can't call For multiple times on one builder
-		Watches(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForObject{}).
-		Owns(&corev1.Secret{}).
-		WithEventFilter(isPullSecret).
+		Watches(
+			&source.Kind{Type: &corev1.Secret{}},
+			&handler.EnqueueRequestForObject{},
+			builder.WithPredicates(pullSecretPredicate),
+		).
+		Owns(&corev1.Secret{}, builder.WithPredicates(pullSecretPredicate)).
```

Comment:
We do not seem to set controller refference on these objects, so I believe that this makes no effect.